src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/BlockPlacer.java

```diff
@@ -53,6 +57,23 @@ public AutonomousMachineHandler getItemHandler() {
 					}
 					else {
 						block.setType(e.getItem().getType());
+						if(e.getItem().hasItemMeta() && (e.getItem().getItemMeta() instanceof BlockStateMeta)) {
+							BlockState itemBlockState = ((BlockStateMeta) e.getItem().getItemMeta()).getBlockState();
+							BlockState blockState = block.getState();
+							
+							if(blockState instanceof Nameable) {
```

Comment:
Should probably also check for hasDisplayName on the ItemMeta first

Reply:
Will do.

Reply:
Ah, please add a comment to the code explaining that then.